src/signal.c

```diff
@@ -15,113 +16,133 @@
 
 #include <libswiftnav/signal.h>
 
-typedef struct {
-  u8 constellation;
-  u8 band;
-  u16 sat_count;
-  u16 (*sat_get)(u32 local_index);
-  u32 (*local_index_get)(u16 sat);
-} sat_group_t;
+/** \defgroup signal GNSS signal identifiers (SID)
+ * \{ */
 
-static const char * constellation_strs[CONSTELLATION_COUNT] = {
-  [CONSTELLATION_GPS] = "GPS",
-  [CONSTELLATION_SBAS] = "SBAS"
-};
+/** Macro used to assert that the specified code is valid. */
+#define ASSERT_CODE_VALID(code) assert(((code) >= 0) && ((code) < CODE_COUNT))
 
-static const char * band_strs[BAND_COUNT] = {
-  [BAND_L1] = "L1"
+/** Element in the code data table. */
+typedef struct {
+  enum constellation constellation;
+  u16 sat_count;
+  u16 sat_start;
+  const char *str;
+} code_table_element_t;
+
+/** Table of useful data for each code. */
+static const code_table_element_t code_table[CODE_COUNT] = {
+  [CODE_GPS_L1CA] =
+    {CONSTELLATION_GPS, NUM_SIGNALS_GPS_L1CA, GPS_FIRST_PRN, "GPS L1CA"},
```

Comment:
Yeah, agree. I do think it's better as is than changing it to signal_count and making the assumption that signal_count is equivalent to sat_count in the code.